src/components/expression/expression-renderer.js

```diff
@@ -518,53 +674,93 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
-      }.bind(this))
-      .on('click', function() {
-        var conditionIndex = Math.floor(xScale.invert(
-            d3.mouse(d3.event.target)[0]) + 0.5);
-        var value = heatmapData.values[geneIndex][conditionIndex];
-        _(this.data.profiles[i]).extend({
-          container_: d3.event.target,
-          hoverColumn: conditionIndex,
-          hoverConditionName: heatmapData.conditionNames[conditionIndex],
-          hoverValue: value
-        });
-        this.signal('pathClick', this.data.profiles[i]);
-      }.bind(this));
-  }, this);
+
+  var legendRect = this.svgLegend_.selectAll('rect')
+    .data(this.data.profiles);
+  legendRect.enter().append('rect');
+  legendRect
+    .attr('height', legendHeight)
+    .attr('width', legendHeight)
+    .attr('x', function(profile, i) {
+      return i * (legendHeight +
+        profile.geneName.length * this.GENE_LABEL_WIDTH_FACTOR);
+    }.bind(this))
+    .style('fill', function(profile) {
+      return this.COLOR_CATEGORY[genotet.utils.hashString(profile.geneName) %
+        this.COLOR_CATEGORY_SIZE];
+    }.bind(this));
+  legendRect.exit().remove();
+
+  var legendText = this.svgLegend_.selectAll('text')
+    .data(this.data.profiles);
+  legendText.enter().append('text');
+  legendText
+    .text(function(profile) {
+      return profile.geneName;
+    })
+    .attr('transform', genotet.utils.getTransform([
+      legendHeight + this.HEATMAP_LEGEND_MARGIN,
+      legendHeight / 2 + this.TEXT_HEIGHT / 2
+    ]))
+    .attr('x', function(profile, i) {
+      return i * (legendHeight +
+        profile.geneName.length * this.GENE_LABEL_WIDTH_FACTOR);
+    }.bind(this));
+  legendText.exit().remove();
+
+  var profilePath = this.profileContent_.selectAll('path')
+    .data(this.data.profiles);
+  profilePath.enter().append('path');
+  profilePath
+    .classed('profile', true)
+    .attr('d', function(profile) {
+      return line(heatmapData.values[profile.row]);
+    })
+    .attr('transform', genotet.utils.getTransform([
+      this.heatmapWidth_ / (heatmapData.conditionNames.length * 2),
+      0
+    ]))
+    .style('stroke', function(profile, i) {
+      var pathColor = this.COLOR_CATEGORY[genotet.utils.hashString(
+        profile.geneName) % this.COLOR_CATEGORY_SIZE];
+      this.data.profiles[i].color = pathColor;
+      return pathColor;
+    }.bind(this))
+    .on('mousemove', function(profile, i) {
+      var conditionIndex = Math.floor(
+        xScale.invert(d3.mouse(d3.event.target)[0]) + 0.5
+      );
+      var value = heatmapData.values[profile.row][conditionIndex];
+      _(this.data.profiles[i]).extend({
+        container_: d3.event.target,
+        hoverColumn: conditionIndex,
+        hoverConditionName: heatmapData.conditionNames[conditionIndex],
+        hoverValue: value
+      });
+      this.signal('pathHover', this.data.profiles[i]);
+    }.bind(this))
+    .on('mouseout', function(profile, i) {
+      this.data.profiles[i].container_ = d3.event.target;
```

Comment:
You shouldn't be accessing a private variable like this. Either make it public or create a wrapper set function.
```js
genotet.ExpressionRenderer.Profile.prototype.setContainer = function(container) {
  this.container_ = container;
};
```